src/Player/PlayerState/PlayerStateHack.cpp

```diff
@@ -0,0 +1,276 @@
+#include "Player/PlayerState/PlayerStateHack.h"
+#include "Player/PlayerHackKeeper.h"
+#include "Player/PlayerAnimator.h"
+#include "Player/IPlayerModelChanger.h"
+#include "Player/HackCap.h"
+#include "Player/CapFunction.h"
+#include <al/Library/Nerve/NerveSetupUtil.h>
+#include <al/Library/Effect/EffectSystemInfo.h>
+#include <al/Library/LiveActor/LiveActor.h>
+#include <al/Library/Nerve/NerveUtil.h>
+#include <al/Library/LiveActor/ActorPoseKeeper.h>
+#include <al/Library/LiveActor/ActorMovementFunction.h>
+#include <al/Library/LiveActor/ActorFlagFunction.h>
+#include <al/Library/LiveActor/ActorActionFunction.h>
+#include <al/Library/LiveActor/ActorSensorFunction.h>
+#include <al/Library/Math/MathUtil.h>
+#include <al/Library/Camera/CameraUtil.h>
+
+namespace {
+    NERVE_IMPL(PlayerStateHack, End);
+    NERVE_IMPL(PlayerStateHack, Hack);
+    NERVE_IMPL(PlayerStateHack, HackDemo);
+    NERVE_IMPL(PlayerStateHack, HackDemoPuppetable);
+    NERVE_MAKE(PlayerStateHack, End);
+    struct {
+        NERVE_MAKE(PlayerStateHack, Hack);
+        NERVE_MAKE(PlayerStateHack, HackDemoPuppetable);
+        NERVE_MAKE(PlayerStateHack, HackDemo);
+    } nrvPlayerStateHack;
+}
+
+PlayerStateHack::PlayerStateHack(al::LiveActor *parent, PlayerHackKeeper *hackKeeper, IPlayerModelChanger *modelChanger,
+                                 PlayerAnimator *animator, HackCap *hackCap) : ActorStateBase("ひょうい", parent),
+                                 mHackKeeper(hackKeeper), mModelChanger(modelChanger), mAnimator(animator),
+                                 mHackCap(hackCap) {
+    mIsStageStartHack = false;
+    mNewSensorTrans = {0.0f, 0.0f, 0.0f};
+    mSensorTrans = {0.0f, 0.0f, 0.0f};
+    mTrans = {0.0f, 0.0f, 0.0f};
+    mHackDemoStartLength = 0;
+    mPossessTraceMtx = sead::Matrix34f::ident;
+    mDemoModelMtx = sead::Matrix34f::ident;
+    initNerve(&End, 0);
+    al::setEffectFollowMtxPtr(parent, "PossessTrace", &mPossessTraceMtx);
+}
+
+void PlayerStateHack::appear() {
+    setDead(false);
+    if (mIsStageStartHack) {
+        mIsStageStartHack = false;
+        al::setNerve(this, &nrvPlayerStateHack.Hack);
+        return;
+    }
+    if (mHackKeeper->isPuppetable2()) {
+        al::setNerve(this, &nrvPlayerStateHack.HackDemoPuppetable);
+        return;
+    }
+    al::setNerve(this, &nrvPlayerStateHack.HackDemo);
+}
+
+void PlayerStateHack::exeEnd() {
+
+}
+
+void PlayerStateHack::exeHack() {
+    if (mHackKeeper->isHack())
+        return;
+    al::copyPose(getActor(), mHackKeeper->getHack());
+    al::setVelocity(getActor(), al::getVelocity(mHackKeeper->getHack()));
+}
+
+void PlayerStateHack::exeHackDemo() {
```

Comment:
This overall function should get a few cleanup-passthroughs, it's still quite a mess.